src/Stream/Topic.h

```diff
@@ -25,7 +26,15 @@ class Topic {
   Topic(Kafka::BrokerSettings const &Settings, std::string const &Topic,
         SrcToDst Map, MessageWriter *Writer, Metrics::Registrar &RegisterMetric,
         time_point StartTime, duration StartTimeLeeway, time_point StopTime,
-        duration StopTimeLeeway);
+        duration StopTimeLeeway,
+        std::unique_ptr<Kafka::ConsumerFactoryInterface> CreateConsumers)
+      : KafkaSettings(Settings), TopicName(Topic), DataMap(std::move(Map)),
```

Comment:
I do not see a good reason to putting the constructor implementation in the header file. It is not helpful to the user of this class to have it here I think. Do you have a convincing reference which argues for doing it?

Reply:
In past projects I was encouraged to put trivial, initialiser-list constructors in the header as it doesn't make a difference to the compiler in this case and means you only have to change it in one place. I don't have strong feelings about it, it is just habit. I'd forgotten you prefer the opposite; I'll put it back.

Reply:
done